plugins/civchat2-paper/src/main/java/vg/civcraft/mc/civchat2/CivChat2Manager.java

```diff
@@ -345,6 +381,40 @@ public void addGroupChat(Player player, Group group) {
         scoreboardHUD.updateScoreboardHUD(player);
     }
 
+
+    //method to check if a message contains a banned word
+    private boolean containsBannedWord(String message) {
+    String lowerMessage = message.toLowerCase();
+    
+    // Load banned words from resources
+    List<String> bannedWords = new ArrayList<>();
+    try (InputStream is = getClass().getResourceAsStream("/banned-words.txt");
+         BufferedReader reader = new BufferedReader(new InputStreamReader(is))) {
+        
+        String line;
+        while ((line = reader.readLine()) != null) {
+            String cleanWord = line.trim().toLowerCase();
+            if (!cleanWord.isEmpty() && !cleanWord.startsWith("#")) {
+                bannedWords.add(cleanWord);
+            }
+        }
+    } catch (IOException e) {
+        System.err.println("Error reading banned words from resources: " + e.getMessage());
+        return false;
+    }
+    
+    System.out.println("Loaded banned words: " + bannedWords);
+    
+    String[] words = lowerMessage.split("\\s+|[.,!?;:]");
```

Comment:
You could do a contains on the entire word. It would be more likely to have false positives but if we don't have that many banned words it should be fine